### PR TITLE
[docs] Add doc sections / pages for upcoming features on pickers

### PR DESCRIPTION
**Pull request by flaviendelangle**
If you have other ideas to add 
The goal being to start gathering feedbacks about the big features the community want (both in the community and the pro plan).

### CODE REVIEW

File: docs/data/date-pickers/date-range-picker/date-range-picker.md [hunk 1]
```diff
@@ -68,3 +68,11 @@ The displayed days are customizable with the `renderDay` function prop.
...
 
 {{"demo": "CustomDateRangePickerDay.js"}}
+
+## 🚧 Pre-defined range shortcuts
+
+> ⚠️ This feature isn't implemented yet. It's coming.
+>
+> 👍 Upvote [issue #4563](https://github.com/mui/mui-x/issues/4563) if you want to see it land faster.
+
+Range shortcuts allows your users to select a commonly-used range in one click (eg: last week, last month, ...)
```
Comment (oliviertassinari) [inline docs/data/date-pickers/date-range-picker/date-range-picker.md:78]:
```suggestion
Range shortcuts allows your users to select a commonly-used range in one click (eg: last week, last month, …)
```

File: docs/data/date-pickers/time-range-picker/time-range-picker.md [hunk 2]
```diff
@@ -0,0 +1,18 @@
+---
+product: date-pickers
```
Comment (oliviertassinari) [inline docs/data/date-pickers/time-range-picker/time-range-picker.md:2]: Why is the product not MUI X?

```suggestion
product: mui-x
```

File: docs/data/date-pickers/time-range-picker/time-range-picker.md [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+title: React Time Range Picker component
+components: TimeRangePicker, DesktopTimeRangePicker, MobileTimeRangePicker, StaticTimeRangePicker
+githubLabel: 'component: TimeRangePicker'
+packageName: '@mui/x-date-pickers'
+materialDesign: https://material.io/components/date-pickers
+---
+
+# Time Range Picker [<span class="plan-pro"></span>](https://mui.com/store/items/material-ui-pro/)
```
Comment (oliviertassinari) [inline docs/data/date-pickers/time-range-picker/time-range-picker.md:10]: Current capitalization policy

```suggestion
# Time range picker [<span class="plan-pro"></span>](https://mui.com/store/items/material-ui-pro/)
```

File: docs/data/date-pickers/time-range-picker/time-range-picker.md [hunk 4]
```diff
@@ -0,0 +1,18 @@
+---
+product: date-pickers
+title: React Time Range Picker component
+components: TimeRangePicker, DesktopTimeRangePicker, MobileTimeRangePicker, StaticTimeRangePicker
```
Comment (oliviertassinari) [inline docs/data/date-pickers/time-range-picker/time-range-picker.md:4]: It creates 404 on the footer https://deploy-preview-4603--material-ui-x.netlify.app/x/react-date-pickers/time-range-picker/

```suggestion
```

File: docs/data/date-pickers/time-range-picker/time-range-picker.md [hunk 5]
```diff
@@ -0,0 +1,18 @@
...
+githubLabel: 'component: TimeRangePicker'
+packageName: '@mui/x-date-pickers'
+materialDesign: https://material.io/components/date-pickers
```
Comment (oliviertassinari) [inline docs/data/date-pickers/time-range-picker/time-range-picker.md:7]: Without `{{"component": "modules/components/ComponentLinkHeader.js"}}` this does nothing (also true for the other pages)

**Review by oliviertassinari — COMMENTED** [review]
I have added this label on the linked GitHub issues so we know that we can't compare the upvotes on these issues with the issues that are not linked in the docs

Screenshot 2022-04-26 at 14 18 42

File: docs/data/date-pickers/time-range-picker/time-range-picker.md (hunk 2 above)
Comment (flaviendelangle) [reply docs/data/date-pickers/time-range-picker/time-range-picker.md:2]: I kept it as is during the migration

https://github.com/mui/material-ui/pull/32178/files#r848633431

I asked the same question here :laughing:

File: docs/data/date-pickers/time-range-picker/time-range-picker.md (hunk 5 above)
Comment (flaviendelangle) [reply docs/data/date-pickers/time-range-picker/time-range-picker.md:7]: I never worked with `ComponentLinkHeader`, do you think we should add it or do you think we should remove the useless headers ?

File: docs/data/date-pickers/time-range-picker/time-range-picker.md (hunk 3 above)
Comment (flaviendelangle) [reply docs/data/date-pickers/time-range-picker/time-range-picker.md:9]: I am fine with removing the capital letters
But aren't we using them in similar scenarios ?

https://mui.com/material-ui/material-icons/
https://mui.com/x/react-date-pickers/date-range-picker/#main-content

**Review by alexfauquette — APPROVED** [review]
Maybe providing another time picker then the clock: https://github.com/mui/mui-x/issues/4483

But not sure how to present that in the doc

File: docs/data/pages.ts [hunk 6]
```diff
@@ -110,6 +110,16 @@ const pages = [
       },
       { pathname: '/x/react-date-pickers/date-time-picker' },
       { pathname: '/x/react-date-pickers/time-picker' },
+      {
+        pathname: '/x/react-date-pickers/time-range-picker',
```
Comment (alexfauquette) [inline docs/data/pages.ts:114]:
```suggestion
        pathname: '/x/react-date-pickers/date-time-range-picker',
```

File: docs/data/date-pickers/date-range-picker/date-range-picker.md [hunk 7]
```diff
@@ -68,3 +68,11 @@ The displayed days are customizable with the `renderDay` function prop.
...
 
 {{"demo": "CustomDateRangePickerDay.js"}}
+
+## 🚧 Pre-defined range shortcuts
+
+> ⚠️ This feature isn't implemented yet. It's coming.
+>
+> 👍 Upvote [issue #4563](https://github.com/mui/mui-x/issues/4563) if you want to see it land faster.
+
+Range shortcuts allows your users to select a commonly-used range in one click (eg: last week, last month, …)
```
Comment (joserodolfofreitas) [inline docs/data/date-pickers/date-range-picker/date-range-picker.md:78]:
```suggestion
Range shortcuts allows your users to select a commonly-used range in one click (eg: last week, last month, …)

## 🚧 Month Range Picker

> ⚠️ This feature isn't implemented yet. It's coming.
>
> 👍 Upvote [issue #4995](https://github.com/mui/mui-x/issues/4995) if you want to see it land faster.

Month Range picker allows setting date ranges by picking months only.
```

**Review by joserodolfofreitas — APPROVED** [review]
I'm adding a new issue here and I think we can publish the current list and open new PRs when we want to add others.

**Pull request merged**